tests/suites/cli/display_clouds.sh

```diff
@@ -111,6 +111,90 @@ EOF
 	fi
 }
 
+run_controller_clouds() {
+	echo
+
+	juju add-cloud my-ec2 -f "./tests/suites/cli/clouds/myclouds.yaml" --force --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME}
+	OUT=$(juju clouds --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME} --format=json 2>/dev/null | jq '.[]')
+
+	EXPECTED=$(
+		cat <<'EOF'
+{
+  "defined": "public",
+  "type": "ec2",
+  "auth-types": [
+    "access-key"
+  ],
+  "regions": {
+    "us-west-1": {
+      "endpoint": "https://ec2.us-west-1.amazonaws.com"
+    },
+    "us-west-2": {
+      "endpoint": "https://ec2.us-west-2.amazonaws.com"
+    }
+  },
+  "users": {
+    "admin": {
+      "display-name": "admin",
+      "access": "admin"
+    }
+  }
+}
+EOF
+	)
+	if [[ "${OUT}" != *"${EXPECTED}"* ]]; then
+		echo "expected ${EXPECTED}, got ${OUT}"
+		exit 1
+	fi
+}
+
+run_controller_credentials() {
+	echo
+
+	juju add-credential my-ec2 -f "./tests/suites/cli/clouds/mycreds.yaml" --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME}
+	OUT=$(juju credentials --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME} --format=json 2>/dev/null | jq '.[] ."my-ec2"')
+
+	EXPECTED=$(
+		cat <<'EOF'
+{
+  "cloud-credentials": {
+    "mine": {
+      "auth-type": "access-key",
+      "details": {
+        "access-key": "my-key"
+      }
+    }
+  }
+}
+EOF
+	)
+	if [[ "${OUT}" != *"${EXPECTED}"* ]]; then
+		echo "expected ${EXPECTED}, got ${OUT}"
```

Comment:
considering $EXPECTED and $OUT are both multi-line strings, this is not going to format very well. I would do something like
```bash
echo "obtained cloud credentials don't match expected"
echo "received: ----------------------"
echo $OUT
echo "expected: ----------------------"
echo $EXPECTED
echo "--------------------------------"
```
Or since you do this logic several times, you could put it in a separate method.

Reply:
it doesn't look too bad tbh, i had no problems debugging things and the current code does it all on one line
i'd much rather conform to the existing style used already in this suite; there's too many of these things to fix to shave the yak for every single one